cpp/unreal_projects/SpearSim/Python/validate_spear_scene.py

```diff
@@ -0,0 +1,579 @@
+            if len(material_path) < 4:
+                spear.log(
+                    "WARNING: Unexpected material path: ",
+                    get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                return
+
+            if material_path[0:5] != ("/", "Game", "Scenes", level, "Materials") and \
+                    material_path[0:3] != ("/", "Game", "Megascans") and \
+                    material_path[0:4] != ("/", "Game", "Kujiale", "Materials"):
+                spear.log(
+                    "WARNING: Unexpected material path: ",
+                    get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                return
+
+            base_material = material.get_base_material()
+            if base_material.get_name() == "M_BaseMaterial":
+                if material_path[-1] != "M_BaseMaterial.M_BaseMaterial" and not material_path[-1].startswith("MI_"):
+                    spear.log(
+                        "WARNING: Unexpected material path: ",
+                        get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                    return
+
+            if base_material.get_name() == "M_TranslucentMaterial":
+                if material_path[-1] != "M_TranslucentMaterial.M_TranslucentMaterial" and not material_path[-1].startswith("MI_"):
+                    spear.log(
+                        "WARNING: Unexpected material path: ",
+                        get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                    return
+
+
+def vector_to_numpy(vector):
+    return np.array([vector.x, vector.y, vector.z])
+
+
+# Unlike the default ActorComponent.get_parent_components() function, this function returns parent components in root-to-leaf order.
+def get_parent_components(component):
+    c = component
+    parents = []
+    while c.get_attach_parent() is not None:
+        c = c.get_attach_parent()
+        parents = [c] + parents
+    return parents
+
+
+def get_debug_string_static_mesh(static_mesh):
+    return static_mesh.get_path_name()
+
+
+def get_debug_string_material(material):
+    return material.get_path_name()
+
+
+def get_debug_string_component(component):
+    return ".".join([c.get_name() for c in get_parent_components(component)]) + "." + component.get_name()
+
+
+def get_debug_string_type(type):
+    return type.__name__
+
+
+def get_debug_string_actor(actor):
+    return str(actor.get_folder_path()) + posixpath.sep + actor.get_actor_label()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--level", default="")
+    args = parser.parse_args()
+
+    level_editor_subsystem = unreal.get_editor_subsystem(unreal.LevelEditorSubsystem)
```

Comment:
Move `level_editor_subsystem` inside the if statement because it is only used there.